### ISSUE TITLE
If condition doesn't recognize that the value is a function

### ISSUE THREAD

**Issue by AndreuCodina** [astral-sh/ty#2950]
### Summary

Recently, I had a deployment with unexpected behaviors because we have a function to detect if the environment is productive, and then add or not observability, etc.

```python
# https://play.ty.dev/10e66926-b958-4685-b37f-a040f6e5315c

def is_production() -> bool:
    return True

if is_production:
    pass

if is_production():
    pass
```

In Pyright, I get the next error message:
Conditional expression references function which always evaluates to True (Pylance [reportUnnecessaryComparison](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportUnnecessaryComparison.md)).

### Version

0.0.20

**Comment by AlexWaygood** [comment]
Thanks! This is tracked in #1073
